src/kaboom.ts

```diff
@@ -5467,11 +5477,335 @@ export default (gopt: KaboomOpt = {}): KaboomCtx => {
+				if (path) {
+					return [from, ...path.slice(1, -1).map(tilePos => tilePos.scale(tw, th).add(tw / 2, th / 2)), to]
+				}
+				else {
+					return null
+				}
+			},
+	
+			/**
+			 * Private methods start here
+			 */
+	
+			_getNeighbours(this: GameObj<LevelComp | PathfindingComp>, node: number, diagonals?: boolean) {
+				const numColumns = this.numColumns()
+				const numRows = this.numRows()
+				const n = []
+				const x = Math.floor(node % numColumns)
+				const left = x > 0 &&
+					(_edgeMap[node] & EdgeMask.Left) &&
+					_costMap[node - 1] !== Infinity
+				const top = node >= numColumns &&
+					(_edgeMap[node] & EdgeMask.Top) &&
+					_costMap[node - numColumns] !== Infinity
+				const right = x < numColumns - 1 &&
+					(_edgeMap[node] & EdgeMask.Right) &&
+					_costMap[node + 1] !== Infinity
+				const bottom = node < numColumns * numRows - numColumns - 1 &&
+					(_edgeMap[node] & EdgeMask.Bottom) &&
+					_costMap[node + numColumns] !== Infinity
+				if (diagonals) {
+					if (left) {
+						if (top) { n.push(node - numColumns - 1) }
+						n.push(node - 1)
+						if (bottom) { n.push(node + numColumns - 1) }
+					}
+					if (top) {
+						n.push(node - numColumns)
+					}
+					if (right) {
+						if (top) { n.push(node - numColumns + 1) }
+						n.push(node + 1)
+						if (bottom) { n.push(node + numColumns + 1) }
+					}
+					if (bottom) {
+						n.push(node + numColumns)
+					}
+				}
+				else {
+					if (left) {
+						n.push(node - 1)
+					}
+					if (top) {
+						n.push(node - numColumns)
+					}
+					if (right) {
+						n.push(node + 1)
+					}
+					if (bottom) {
+						n.push(node + numColumns)
+					}
+				}
+				return n
+			},
+	
+			_getCost(this: GameObj<LevelComp | PathfindingComp>, node: number, neighbour: number) {
+				// Cost of destination tile
+				return _costMap[neighbour]
+			},
+	
+			_getHeuristic(this: GameObj<LevelComp | PathfindingComp>, node: number, goal: number) {
+				// Euclidian distance to target
+				const p1 = this._hash2Tile(node)
+				const p2 = this._hash2Tile(goal)
+				return p1.dist(p2)
+			},
+	
+			// The obstacle map tells which tiles are accessible
+			// Cost: accessible with cost
+			// Infinite: inaccessible
+			_createCostMap(this: GameObj<LevelComp | PathfindingComp>) {
+				const spatialMap = this.getSpatialMap()
+				const size = this.numRows() * this.numColumns()
+				if (!_costMap) {
+					_costMap = new Array<number>(size)
+				}
+				else {
+					_costMap.length = size
+				}
+				_costMap.fill(1, 0, size)
+				for (let i = 0; i < spatialMap.length; i++) {
+					const objects = spatialMap[i]
+					if (objects) {
+						const len = objects.length
+						let cost = 0
+						for (let j = 0; j < len; j++) {
+							const object = objects[j]
+							if (object.isObstacle !== undefined) {
+								if (object.isObstacle) {
+									cost = Infinity
+									break
+								}
+							}
+							else {
+								cost += object.cost || 0
+							}
```

Comment:
Is it ok to simplify it to
```js
cost += object.isObstacle ? Infinity : (object.cost ?? 0)
```

Reply:
You would need at least

cost += object.isObstacle !== undefined ? (object.isObstacle ? Infinity : 0) : (object.cost ?? 0)
And you'd lose the early exit

Reply:
You meant early exit for `object.isObstacle !== undefined`? Yeah cuz I was also wondering if `isObstacle = false` tile can also have a `cost`

Reply:
A user could use both cost and obstacle yes

if (object.isObstacle !== undefined && object.isObstacle) {
    cost = Infinity;
    break;
}
else {
    cost += object.cost || 0;
}

Reply:
Oh you meant the early `break` sorry

Reply:
Yes, once you find an obstacle, you don't need to check other objects anymore.